orthoinference/src/main/java/org/reactome/orthoinference/StableIdentifierGenerator.java

```diff
@@ -0,0 +1,67 @@
+package org.reactome.orthoinference;
+
+import org.gk.model.GKInstance;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.HashMap;
+import java.util.Map;
+import static org.gk.model.ReactomeJavaConstants.*;
+
+/*
+ *  All PhysicalEntitys, ReactionlikeEvents and Pathways are routed to this class to generate their stable identifiers
+ */
+public class StableIdentifierGenerator {
+
+    private static MySQLAdaptor dba;
+    private static GKInstance instanceEditInst;
```

Comment:
The `instanceEditInst` doesn't seem to be used.  Can it be added to the modified attribute after the generated stable id is added to the inferred instance?

Reply:
Also as the created attribute for the new stable id instance?

Reply:
Right, the 'created' attribute is given to it in the InstanceUtilities call, the instance edit survived from when I was doing the code seperate

Reply:
Ah okay, great.  Does the modified attribute need to be updated for the inferred instance after it gets the stable identifier added to it on line 49?

Reply:
I think this is a philosophical question. The StableIdentifier instance will now be created right before the inferred instance, which means that the inferred instance will already have the stable identifier attribute populated by the time its stored in the database. Does it make sense to denote a 'modify' for an instance when its now a part of the creation process?

Reply:
Oh, I see.  I thought the inferred instance was created and stored at this point, but if not then no need for a modified value!

Is there any chance that the inferred instance won't be stored because of an aborted reaction or pre-existing identical instance after the stable identifier instance is already created and stored?

Reply:
I positioned the stable identifier generator right before the 'store' method is called on the dba for the instance in question. See:
[PhysicalEntitys](https://github.com/reactome/data-release-pipeline/blob/feature/add_ortho_stableIds/orthoinference/src/main/java/org/reactome/orthoinference/InstanceUtilities.java#L136), [Reactions](https://github.com/reactome/data-release-pipeline/blob/feature/add_ortho_stableIds/orthoinference/src/main/java/org/reactome/orthoinference/ReactionInferrer.java#L93), [Pathways](https://github.com/reactome/data-release-pipeline/blob/feature/add_ortho_stableIds/orthoinference/src/main/java/org/reactome/orthoinference/HumanEventsUpdater.java#L141)

So while obviously there is always a chance, I think that if a Stable Identifier instance is being created it should mean that its referring instance will be inserted

Reply:
Okay, great.  Yeah, that makes sense.  From a code logic standpoint, it shouldn't happen but an interruption of running could cause it.

In that case, what would happen if the stable id got created and stored but the process was interrupted before storing the inferred instance?  When re-running and resuming creating the inferred instance would it find the already created stable identifier instance and attach it or would it create a new stable identifier instance?

Reply:
It would create a new one, since the Stable Identifier code doesn't do any identical instance checks. I can't really think of a way around that situation though that wouldn't cause a notable runtime increase, as the identical instance checks are the expensive ones

Reply:
We could try searching for a stable identifier with the same identifier attribute value.  Maybe a cached set of all existing stable identifiers with the species prefix so it's just one DB look-up?

Reply:
If an interruption happens, should the database be restored and the process restarted from scratch, or is the process generally robust enough to survive an abort and be safely re-run on the same database as before the interruption?

Reply:
I think it should usually be robust enough to continue.  If the interruption happens because of a Reactome data related problem rather than a system one, though, it may be better to start from scratch.

Reply:
Yea, I'm going to create an issue of this just as a reminder to test that the script is interruption proof

Reply:
Done